keps/sig-node/4680-add-resource-health-to-pod-status/README.md

```diff
@@ -0,0 +1,453 @@
+
+##### Prerequisite testing updates
+
+Device Plugin and DRA are relatively new features and have a reasonable test coverage.
+
+##### Unit tests
+
+- `k8s.io/kubernetes/pkg/kubelet/cm/devicemanager`: `5/31/2024` - `84.1`
+- `k8s.io/kubernetes/pkg/kubelet/cm/dra`: `5/31/2024` - `59.2`
+- `k8s.io/kubernetes/pkg/kubelet/cm/dra/plugin`: `5/31/2024` - `34`
+- `k8s.io/kubernetes/pkg/kubelet/cm/dra/state`: `5/31/2024` - `98`
+
+##### Integration tests
+
+N/A
+
+##### e2e tests
+
+Planned tests:
+
+- Device marked unhealthy - the state is reflected in pod status
+- Device marked unhealthy and back to healthy after some time - pod status was changed to unhealthy temporarily
+- Device marked as unhealthy and back to healthy in quick succession - pod status reflects the latest health status
+- Pod failed due to unhealthy device, earlier than device plugin detected it. Pod status is still updated.
+- Pod is in crash loop backoff due to unhealthy device - pod status is updated to unhealthy
+
+Test coverage will be listed once tests are implemented.
+
+- <test>: <link to test coverage>
+
+### Graduation Criteria
+
+#### Alpha
+
+- New field is introduced in Pod Status
+- Feature implemented in Device Manager behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Feature is implemented in DRA behind the feature flag
+- Complete e2e tests coverage
+
+#### GA
+
+- Feedback is collected on usability of the field
+- Example of real-world usage with one of the device plugin. For example, NVIDIA Device Plugin
+
+### Upgrade / Downgrade Strategy
+
+The feature exposes a new field based on information the Device Plugin already exposes. There will be no dependency on upgrade/downgrade, feature will either work or not.
+
+DRA implementation requires DRA interfaces change. DRA is in alpha and in active development. The feature will follow the DRA ugrade/downgrade strategy.
+
+### Version Skew Strategy
+
+There is no issue with the version skew. Kubelet that will expose this flag will
+always be the same version of behind the API, which introduced this new field.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+Simple change of a feature gate will either enable or disable this feature.
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `ResourceHealthStatus`
+  - Components depending on the feature gate: `kubelet` and `kube-apiserver`
+
+###### Does enabling the feature change any default behavior?
+
+No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, with no side effect except of missing the new field in pod status.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The pod status will be updated again.
+
+###### Are there any tests for feature enablement/disablement?
+
+Nothing is planned.
```

Comment:
This is required for alpha for features with new fields like this.  For this KEP update, let's list out what the rough test coverage should be. 

Here's an example of the cases we typcally cover, from a simple field addition: https://github.com/kubernetes/kubernetes/pull/122717/files#diff-c4fc49b3c4f7fe0c2f6aa7c99c42cd28d1234efe3c143d05dd126808714b155bR700-R1298

Reply:
Added to the e2e section